var/spack/repos/builtin/packages/ruby/package.py

```diff
@@ -115,14 +115,15 @@ def post_install(self):
                                             'ssl_certs')
             install(rubygems_updated_cert_path, rubygems_certs_path)
 
-        rbconfig = '{0}/lib/ruby/{1}.0/x86_64-linux/rbconfig.rb'.format(
-                self.spec.prefix, self.spec.version.up_to(2))
-        filter_file(r'^(\s*CONFIG\["CXX"\]\s*=\s*).*',
-                    '\\1"{0}"'.format(self.compiler.cxx),
-                    rbconfig)
-        filter_file(r'^(\s*CONFIG\["CC"\]\s*=\s*).*',
-                    '\\1"{0}"'.format(self.compiler.cc),
-                    rbconfig)
-        filter_file(r'^(\s*CONFIG\["MJIT_CC"\]\s*=\s*).*',
-                    '\\1"{0}"'.format(self.compiler.cc),
-                    rbconfig)
+        if self.spec.target.family == 'x86_64' and self.spec.platform == 'linux':
+            rbconfig = '{0}/lib/ruby/{1}.0/x86_64-linux/rbconfig.rb'.format(
```

Comment:
Is there only a single directory in this location? Maybe we can just glob and check for all possible `rbconfig.rb` files. I would hate to see a solution that doesn't work on all platforms (as a macOS user myself).

Reply:
I'd really like to do:

```python
rbconfig = '{0}/lib/ruby/{1}.0/{2}-{3}/rbconfig.rb'.format(
    self.spec.prefix,
    self.spec.version.up_to(2),
    self.spec.target.family,
    self.spec.platform)
```
I just don't know if that's valid in all cases.

Reply:
What does that directory structure look like for you when you install it?

Reply:
I'll give it a try this weekend.

Reply:
Okay, here's what I'm seeing:
```console
$ find . -name rbconfig.rb
./lib/ruby/2.7.0/x86_64-darwin19/rbconfig.rb
```
So the version is different from what you expected (I actually installed 2.7.1, not sure why it used 2.7.0. What if we do something like the following:
```python
rbconfig = find(self.prefix, 'rbconfig.rb')[0]
```
This will work regardless of what weird things Ruby decides to throw at us.